src/pybind/mgr/prometheus/module.py

```diff
@@ -819,6 +820,32 @@ def _setup_static_metrics(self) -> Dict[str, Metric]:
 
         return metrics
 
+    def get_server_addr(self) -> str:
+        """
+        Return the current mgr server IP.
+        """
+        server_addr = cast(str, self.get_localized_module_option('server_addr', get_default_addr()))
+        if server_addr in ['::', '0.0.0.0']:
+            return self.get_mgr_ip()
+        return server_addr
+
+    def config_notify(self) -> None:
+        """
+        This method is called whenever one of our config options is changed.
+        """
+        # https://stackoverflow.com/questions/7254845/change-cherrypy-port-and-restart-web-server
+        # if we omit the line: cherrypy.server.httpserver = None
+        # then the cherrypy server is not restarted correctly
+        self.log.info('Restarting engine...')
+        cherrypy.engine.stop()
+        cherrypy.server.httpserver = None
+        server_port = cast(int, self.get_module_option_ex(
+            'prometheus', 'server_port', DEFAULT_PORT))
```

Comment:
is there a reason we can't use `self.get_localized_module_option` here the same way we do for `server_addr` in `get_server_addr`

Reply:
Fixed :+1: